### DISCUSSION TITLE
Apostrophe 3.61.0 Astro integration, schema field improvements, and fixes

### DISCUSSION THREAD

**Discussion by BoDonkey** [apostrophecms/apostrophe#4379] [Release Notes]
Hello JavaScript Aficionados!
As the sun sets on the shortest day of the year here at Apostrophe Central, we're thrilled to bring you a special, snug-sized update – just in time for the solstice!

The big news for this release is a new extension that allows you to utilize the very popular [Astro](https://astro.build/) as a front-end for your Apostrophe project without losing any of the ApostropheCMS editing experience! Development of this module began with Stéphane Maccari and Clément Ravier of Michelin. We are grateful for their generous support of ApostropheCMS. We're also releasing two brand-spanking-new templates to get you up and running quickly. The [`starter-kit-astro`](https://github.com/apostrophecms/starter-kit-astro) can be used to create projects with our CLI tool and provides a minimal Apostrophe project starting point for working with Astro. The [`astro-frontend`](https://github.com/apostrophecms/astro-frontend) template provides a minimal Astro project that complements the starter kit beautifully. We recommend having a look at the [README.md](https://apostrophecms.com/extensions/astro-integration) file of the new extension to get all the details. So, grab a cup of your favorite holiday drink, check out these new toys, and let your creativity soar into the new year!

In addition to some key changes that were needed to support Astro integration, the Apostrophe core also got some improvements and fixes. You can now add an `autocomplete` attribute to your schema fields - this can be used to direct the browser as to the type of data a field should receive for autofill or turn off autocompletion for sensitive fields. You can now use `pattern` to restrict what a user can add to a `url` field like you already can with a `string` field. While we were under the hood we also fixed some issues with the image modal in the rich-text-widget and the `if` conditional.

As we all wrap up the year (and some of us literally wrap up presents), we want to extend a heartfelt thank you to our incredible community around the world. Your support and feedback are the gifts that keep on giving, all year long. So, go ahead, dive into our little holiday treat. Whether you're basking in the sun or bundled up by the fire, we hope it adds a dash of joy to your celebrations. Here's to longer days ahead (or shorter, if that's your thing) and to all the adventures 2024 will bring!

Remember `npm update` gets you the latest and greatest right in your project and is *critical* if you want to enjoy the new Astro integration.

### Adds

* Add a `validate` method to the `url` field type to allow the use of the `pattern` property.
* Add `autocomplete` attribute to schema fields that implement it (cf. [HTML attribute: autocomplete](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/autocomplete)).
* Add the `delete` method to the `@apostrophecms/cache` module so we don't have to rely on direct MongoDB manipulation to remove a cache item.
* Adds tag property to fields in order to show a tag next to the field title (used in advanced permission for the admin field). Adds new sensitive label color.
* Pass on the module name and the full, namespaced template name to external front ends, e.g. Astro. Also make this information available to other related methods for future and project-level use.
* Fixes the AposCheckbox component to be used more easily standalone, accepts a single model value instead of an array.

### Fixes

* Fix `date` schema field query builder to work with arrays.
* Fix `if` on pages. When you open the `AposDocEditor` modal on pages, you now see an up to date view of the visible fields.
* Pass on complete annotation information for nested areas when adding or editing a nested widget using an external front, like Astro.
* We can now close the image modal in rich-text widgets when we click outside of the modal. The click on the cancel button now works too.
* Fixes the `clearLoginAttempts` method to work with the new `@apostrophecms/cache` module `delete` method.

## [`@apostrophecms/apostrophe-astro`](https://apostrophecms.com/extensions/astro-integration) v1.0.1
This module integrates ApostropheCMS into your [Astro](https://astro.build/) application.

### 1.0.1 (2023-12-21)

* Fix bug impacting two-column widgets.

### 1.0.0 (2023-12-21)

* Initial release.

**Comment by myovchev** [comment]
Awesome last moment release!
